boot/bootutil/src/loader.c

```diff
@@ -1939,3 +1931,189 @@ split_go(int loader_slot, int split_slot, void **entry)
     free(sectors);
     return rc;
 }
+
+#else /* MCUBOOT_DIRECT_XIP */
+
+/**
+ * Iterates over all slots and determines which contain a firmware image.
+ *
+ * @param state          Boot loader status information.
+ * @param slot_usage     A pointer to a zero-initialized array, which aim is to
+ *                       carry information about slots that contain an image.
+ *                       After return the corresponding array elements are set
+ *                       to a non-zero value if the given slots are in use
+ *                       (contain a firmware image).
+ * @param slot_cnt       The number of slots, which can contain firmware images.
+ *
+ * @return               The number of found images.
+ */
+static uint32_t
+boot_get_slot_usage(struct boot_loader_state *state, uint8_t slot_usage[],
+                    uint32_t slot_cnt)
+{
+    struct image_header *hdr = NULL;
+    uint32_t image_cnt = 0;
+    uint32_t slot;
+
+    for (slot = 0; slot < slot_cnt; slot++) {
+        hdr = boot_img_hdr(state, slot);
+
+        if (boot_is_header_valid(hdr, BOOT_IMG_AREA(state, slot))) {
+            slot_usage[slot] = 1;
+            image_cnt++;
+            BOOT_LOG_IMAGE_INFO(slot, hdr);
+        } else {
+            BOOT_LOG_INF("%s slot: Image not found", (slot == BOOT_PRIMARY_SLOT)
+                         ? "Primary" : "Secondary");
+        }
+    }
+
+    return image_cnt;
+}
+
+int
+context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
+{
+    struct image_header *hdr = NULL;
+    struct image_header *selected_image_header = NULL;
+    uint8_t slot_usage[BOOT_NUM_SLOTS] = {0};
```

Comment:
initialization of array to zeros like that is compiler not mandatory extension.
Better to use memset(), or
rework boot_get_slot_usage() so it clear or set fields.

Reply:
Fixed.

Reply:
thanks, it looks good now.